modin/data_management/functions/binary_function.py

```diff
@@ -11,19 +11,61 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
+from typing import Callable
 import numpy as np
 import pandas
 
 from .function import Function
 
 
 class BinaryFunction(Function):
+    """Builder class for Binary functions."""
+
     @classmethod
-    def call(cls, func, *call_args, **call_kwds):
-        def caller(query_compiler, other, *args, **kwargs):
+    def register(cls, func: Callable, join_type="outer", preserve_labels=False):
```

Comment:
this change is just spreading kwargs because it hides actual function arguments and makes it unclear whether all parameters are documented or not. 

Would doc-string for this function be valid if I keep it as it is and revert the changes in function signature to:
```python
def register(cls, func, **kwargs)
```

Reply:
Oh, noticed that this change affects not only the function argument but also a function name. I'll rename it back, but the above question about kwargs is still relevant

Reply:
Please leave a `#TODO` comment about unpacking `**kwargs`. This PR should only touch docstrings, comments and `.rst` documentation, nothing else.